### PR TITLE
[UNDERTOW-2141] Fix NPE and add a test for it

### PR DESCRIPTION
**Pull request by wal-jan**
Jira: https://issues.redhat.com/browse/UNDERTOW-2141

### CODE REVIEW

File: servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java:3]: Please, change the year to 2022.

File: servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import io.undertow.testutils.HttpClientUtils;
+import io.undertow.testutils.TestHttpClient;
+import io.undertow.util.StatusCodes;
+
+/**
+ *
+ * Test that separate servlet deployments use seperate session managers, even if one is broken.
+ *
+ * @author Stuart Douglas
+ */
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java:53]: You can add your name as author as well if you wish

**Review by fl4via — APPROVED** [review]

File: servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java (hunk 2 above)
Comment (wal-jan) [reply servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java:53]: Thanks for your hint, but imho `git blame` is sufficent

File: servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java (hunk 1 above)
Comment (wal-jan) [reply servlet/src/test/java/io/undertow/servlet/test/session/CrossContextServletBrokenSessionFactoryTestCase.java]: PR updated

**Pull request merged**